lib/content-services/src/lib/agent/services/agent.service.ts

```diff
@@ -36,39 +38,88 @@ export class AgentService {
         this._mocked = mocked;
     }
 
+    agentsList$ = this.agentsList.asObservable();
+
     constructor(private apiService: AlfrescoApiService) {}
 
+    /**
+     * Gets all agents from cache. If cache is empty, fetches agents from backend.
+     *
+     * @returns Agent object containing the agents.
+     */
+    getAgents(): Observable<AgentWithAvatar[]> {
+        return this.agentsList$.pipe(
+            switchMap((agentsList: AgentWithAvatar[]) => {
```

Comment:
no need to set type here, type is detected based on agentsList$ type so can be removed: 
<img width="616" alt="image" src="https://github.com/user-attachments/assets/ac930377-d938-4343-a645-2ae09cc821ff">